packages/blockchain-extension/extension/commands/associateTestDataDirectoryCommand.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+'use strict';
+import * as vscode from 'vscode';
+import { VSCodeBlockchainOutputAdapter } from '../logging/VSCodeBlockchainOutputAdapter';
+import { FabricGatewayRegistryEntry, FabricGatewayRegistry } from 'ibm-blockchain-platform-common';
+import { IBlockchainQuickPickItem, UserInputUtil } from './UserInputUtil';
+import { LogType } from 'ibm-blockchain-platform-common';
+import { InstantiatedTreeItem } from '../explorer/model/InstantiatedTreeItem';
+import { ContractTreeItem } from '../explorer/model/ContractTreeItem';
+import { FabricGatewayConnectionManager } from '../fabric/FabricGatewayConnectionManager';
+import { ExtensionCommands } from '../../ExtensionCommands';
+
+export async function associateTestDataDirectory(chaincode?: InstantiatedTreeItem | ContractTreeItem): Promise<any> {
+    let gateway: FabricGatewayRegistryEntry;
+    let chosenChaincode: IBlockchainQuickPickItem<{ name: string, channel: string, version: string }>;
+    let chaincodeName: string;
+    let chaincodeLabel: string;
+    const outputAdapter: VSCodeBlockchainOutputAdapter = VSCodeBlockchainOutputAdapter.instance();
+    outputAdapter.log(LogType.INFO, undefined, 'associateTestDataDirectory');
+
+    // If called from the command palette, ask for instantiated smart contract to test
+    if (!chaincode) {
+        if (!FabricGatewayConnectionManager.instance().getConnection()) {
+            // Connect if not already connected
+            await vscode.commands.executeCommand(ExtensionCommands.CONNECT_TO_GATEWAY);
+            if (!FabricGatewayConnectionManager.instance().getConnection()) {
+                // either the user cancelled or there was an error so don't carry on
+                return;
+            }
+        }
+
+        // Ask for instantiated smart contract
+        chosenChaincode = await UserInputUtil.showClientInstantiatedSmartContractsQuickPick('Please choose instantiated smart contract associate a test data directory with');
```

Comment:
the prompt doesn't make sense maybe something like `Please choose an instantiated smart contract to associate a transaction data directory with`